### ISSUE TITLE
scaffolder action fetch:template does not preserve file permissions

### ISSUE THREAD

**Issue by twonds** [backstage/backstage#7021]
When creating a template using this action file permissions in github are not kept. Particularly, executable files are just read write when the new repository is created.
 
## Expected Behavior

Permissions in the template's git repository match the output repository 

## Current Behavior

Permissions are not preserved when creating a project from a template

## Possible Solution
I was gonna create a pull request but I need advice on how to write the tests. If this is a possible solution, how best to test this? 

https://github.com/twonds/backstage/commit/652a2e9276e0e359aaaf7339fd4dc02510a7169c
```
diff --git a/plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts b/plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts
index 425623d15..13dfb2486 100644
--- a/plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts
+++ b/plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts
@@ -262,8 +262,9 @@ export function createFetchTemplateAction(options: {
             );
             await fs.copy(inputFilePath, outputPath);
           } else {
+            const statsObj = await fs.stat(inputFilePath);
             ctx.logger.info(
-              `Writing file ${location} to template output path.`,
+              `Writing file ${location} to template output path. mode: ${statsObj.mode}`,
             );
             const inputFileContents = await fs.readFile(inputFilePath, 'utf-8');
             await fs.outputFile(
@@ -272,6 +273,7 @@ export function createFetchTemplateAction(options: {
                 ? templater.renderString(inputFileContents, context)
                 : inputFileContents,
             );
+            await fs.chmod(outputPath, statsObj.mode);
           }
         }
       }
```

## Steps to Reproduce

1. Create template repository with executable files that are not binary
2. Create project from this template in backstage using `fetch:template` action
3. Newly created project's files do not have executable files

## Context
If a project wants to have a grade wrapper or some script wrapper that must be executable then the template will fail to create it.

## Your Environment

The docker image in backstage backend and macos but I believe this happens in any environment.

**Comment by freben** [comment]
Yep, as mentioned on discord, this sounds like it makes a lot of sense. I don't know if there might be any ill effects on windows or other non-posix systems; it may be the case that the stat+chmod should be run in a best-effort try-catch that just silently ignores issues? I am not 100% sure honestly, just thinking out loud

**Comment by mtlewis** [comment]
Thanks for flagging this and offering to contribute the fix! The tests for fetch:template are written using [mock-fs](https://github.com/tschaub/mock-fs), which provides a mock file system that the action reads from during the tests. It looks like it's possible to include files with custom flags using [mock.file](https://github.com/tschaub/mock-fs#mockfileproperties).

It also looks like it should be possible to set the permissions when running outputFile by passing a custom options object, so maybe worth exploring that before adding a separate call. See the [fs-extra docs](https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/outputFile.md).

**Comment by twonds** [comment]
Thank you @freben and @mtlewis for your help. I have written tests and manually tested for verification. I will start preparing the pull request with a changelog update. Hopefully get that in soon!

**Comment by Rugvip** [comment]
Fix from #7051 is now released, thank you @twonds!
